### PR TITLE
CAS E2E test

### PR DESCRIPTION
**Pull request by Rohitrajak1807**
#### Small fix in arlon init  to run without any user prompts

### CODE REVIEW

File: testing/e2e_cas_setup.sh [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+
+cat <<EOF > ${credentialsFile}
+[default]
+aws_access_key_id = ${AWS_ACCESS_KEY_ID}
+aws_secret_access_key = ${AWS_SECRET_ACCESS_KEY}
+EOF
+
+chmod 0644 ${configFile} ${credentialsFile}
+
+export dateSuffix=${date “+%F-%H-%M-%S”}
```
Comment (Rohitrajak1807) [inline testing/e2e_cas_setup.sh:140]:
```suggestion
export dateSuffix=$(date “+%F-%H-%M-%S”)
```

File: testing/e2e_cas_setup.sh [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+eksConfig=eksctl_config.yaml
+eksctl create cluster --config-file ${eksConfig} 
+
+aws eks update-kubeconfig --region ${AWS_REGION} --name eksctl-${dateSuffix}
+
+echo waiting for cluster control plane ...
+# 'kubectl version' queries the server version by default
+wait_until "kubectl version &> /dev/null" 2 30
```
Comment (Rohitrajak1807) [inline testing/e2e_cas_setup.sh:148]: good to have, but not necessary

File: testing/e2e_cas_setup.sh [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+
+git_server_port=${GIT_SERVER_PORT}
+if [ -z "${git_server_port}" ]; then
+  git_server_port=8188
```
Comment (Rohitrajak1807) [inline testing/e2e_cas_setup.sh:152]: remember to set service.http.port=8188 when installing gitea release

File: testing/e2e_cas_setup.sh [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+
+if [ ! -d "${GIT_ROOT}" ]; then
+  mkdir ${GIT_ROOT}
+  chmod og+rwx ${GIT_ROOT}
+fi
+
+if ! kubectl get svc gitea-http; then
+  helm repo add gitea-charts https://dl.gitea.io/charts/
+  helm repo update
+  helm install gitea gitea-charts/gitea --set gitea.admin.username="${GIT_USER}" --set gitea.admin.password="${GIT_PASSWORD}" --set service.http.type=LoadBalancer --wait --wait-for-jobs --timeout 10m
```
Comment (Rohitrajak1807) [inline testing/e2e_cas_setup.sh:168]: please check the comment above

File: testing/e2e_cas_setup.sh [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+  if pkill -f "kubectl --namespace default port-forward svc/gitea-http"; then
+    echo "terminated previous port forwarding session for gitea"
+  fi
+  kubectl --namespace default port-forward svc/gitea-http ${git_server_port}:3000 --address 0.0.0.0 &>/dev/null &
```
Comment (Rohitrajak1807) [inline testing/e2e_cas_setup.sh:172]:
```suggestion

```

File: testing/eksctl_config.yaml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  nat:
+    gateway: HighlyAvailable
+  clusterEndpoints:
+    publicAccess:  true
+    privateAccess: true
+
+addons:
+  - name: vpc-cni
+  - name: coredns
+  - name: kube-proxy
```
Comment (Rohitrajak1807) [inline testing/eksctl_config.yaml:25]: missing the ebs addon

File: testing/e2e_cas_setup.sh (hunk 2 above)
Comment (ShaunakJoshi1407) [reply testing/e2e_cas_setup.sh:148]: Yeah just additional validation as such

File: testing/eksctl_config.yaml (hunk 6 above)
Comment (ShaunakJoshi1407) [reply testing/eksctl_config.yaml:25]: Adding it

File: testing/e2e-cas/00-cas/00-assert.yaml [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  namespace: cas-e2e-cluster
+status:
+  ready: true
+---
+apiVersion: core.arlon.io/v1
+kind: CallHomeConfig
+metadata:
+  namespace: cas-e2e-cluster
+status:
+  state: complete
```
[the remark this answers is not in the document]
Comment (Rohitrajak1807) [reply testing/e2e-cas/00-cas/00-assert.yaml:33]: arlon-app, cluster app, bundle app and the profile app
The healthy state for bundle denotes that CAS pod is running

File: testing/e2e-cas/00-cas/00-assert.yaml [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+kind: AWSManagedControlPlane
+metadata:
+  namespace: cas-e2e-cluster
+status:
+  ready: true
+---
+apiVersion: core.arlon.io/v1
+kind: CallHomeConfig
+metadata:
+  namespace: cas-e2e-cluster
```
Comment (Rohitrajak1807) [inline testing/e2e-cas/00-cas/00-assert.yaml:31]:
```suggestion
  namespace: cas-e2e-cluster
  name: cluster-autoscaler
```

File: testing/e2e_cas_setup.sh [hunk 9]
```diff
@@ -0,0 +1,247 @@
...
+if [[ "$cpu" == "x86_64" ]]; then
+  arlon_arch="x86_64"
+  arch="amd64"
+elif [[ "$cpu" == "arm64"* ]]; then
+  #  arlon_arch="arm64"
+  arch="arm64"
+fi
+if [ ! -d "$HOME/.local/bin" ]; then
+  mkdir -p "$HOME/.local/bin"
+fi
```
Comment (Rohitrajak1807) [inline testing/e2e_cas_setup.sh:84]:
```suggestion
```

Might not be needed

Comment (ShaunakJoshi1407) [reply testing/e2e_cas_setup.sh:84]: We still require this to download kubectl in the setup script

File: testing/e2e-cas/00-cas/00-assert.yaml (hunk 8 above)
Comment (ShaunakJoshi1407) [reply testing/e2e-cas/00-cas/00-assert.yaml:31]: Done

File: testing/e2e_cas_teardown.sh [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+git_server_port=3000
+
+rm -rf ~/.arlon ~/.config/arlon ~/.aws /tmp/arlon*
+workspace_repo="/tmp/arlon-testbed-git-clone/myrepo"
+
+pushd "${workspace_repo}" || exit 1
+git pull
+rm -rf "$(ls)"
+git commit -am "cleanup e2e-test manifests"
+git push "http://${GIT_USER}:${GIT_PASSWORD}@localhost:${git_server_port}/${GIT_USER}/myrepo.git"
```
Comment (Rohitrajak1807) [inline testing/e2e_cas_teardown.sh:32]: not localhost. we need to get the ip

Comment (ShaunakJoshi1407) [reply testing/e2e_cas_teardown.sh]: Removed it

**Review by cruizen — APPROVED** [review]
LGTM

**Comment by Rohitrajak1807** [comment]
There was a delay because of a bug in kuttl, we initiated a conversation for this and it is being fixed. The PR is already out for it and is awaiting reviews. Please check the conversation here: https://kubernetes.slack.com/archives/CG3HTFCMV/p1673420417548399

**Comment by Rohitrajak1807** [comment]
Looks good for now, the only suggestion is to either remove the 01-delete step until the kuttl bug is fixed. It can be preserved in some other branch and merged later perhaps. Apart from that it's good to go.

**Pull request merged**